esx_service/cli/vmdkops_admin.py

```diff
@@ -726,38 +731,382 @@ def get_version():
     except:
         return NOT_AVAILABLE
 
+def get_tenant_from_db(name):
+    try:
+        auth.connect_auth_db()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+    
+    error_info, tenant = auth._auth_mgr.get_tenant(name)
+    return error_info, tenant
+
+def create_tenant_in_db(name, description, default_datastore, default_privileges, vms, privileges):
+    try:
+        auth.connect_auth_db()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+
+    error_info, tenant = auth._auth_mgr.create_tenant(name = name, 
+                                                      description = description, 
+                                                      default_datastore = default_datastore, 
+                                                      default_privileges = default_privileges, 
+                                                      vms = vms, 
+                                                      privileges = privileges)
+    return error_info, tenant
+
+def get_tenant_list_from_db():
+    try: 
+        auth.connect_auth_db()
+        error_info, tenant_list = auth._auth_mgr.list_tenants()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+
+    error_info, tenant_list = auth._auth_mgr.list_tenants()
+    return error_info, tenant_list
+
+def operation_fail(error_info):
+    print error_info
+    return error_info
+
+def tenant_ls_headers():
+    """ Return column names for tenant ls command """
+    headers = ['Uuid', 'Name', 'Description', 'Default_datastore', 'VM_list']
```

Comment:
Default datastore can't be a column header, its one of several datastores that the tenant has entitlements to. Instead list all datastores that the tenant has access to and for the default one mark it as such in its entry.